scripts/BuildNugetPackages/Program.cs

```diff
@@ -0,0 +1,187 @@
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using CommandLine;
+using Medallion.Shell;
+
+namespace BuildNugetPackages
+{
+    internal class Program
+    {
+        private static readonly Shell shell = new Shell(options => options.ThrowOnError());
+
+        private const string CSharpWorkerRepo = "https://github.com/spatialos/csharp-worker-template.git";
+
+        private static int Main(string[] args)
+        {
+            Console.Out.WriteLine(string.Join(" ", args));
+
+            if (args.Length == 0)
+            {
+                // Default to git checkout.
+                args = new[] {"git"};
+
+                var templateBranch = Environment.GetEnvironmentVariable("CSHARP_TEMPLATE_BRANCH");
+
+                if (string.IsNullOrEmpty(templateBranch))
+                {
+                    // Default to "master", unless the downstream dependency has a branch matching the same name, then use that.
+                    var currentBranch = Environment.GetEnvironmentVariable("BUILDKITE_BRANCH");
+
+                    if (string.IsNullOrEmpty(currentBranch))
+                    {
+                        // We're running locally, get the current branch name.
+                        var currentLines = new List<string>();
+                        shell.Run("git", "rev-parse", "--abbrev-ref", "HEAD")
+                            .RedirectTo(currentLines)
+                            .RedirectStandardErrorTo(Console.Error)
+                            .Wait();
+                        currentBranch = currentLines.First().Trim();
```

Comment:
Nity-est of nits: This always isn't a branch 😛

Reply:
Yep, true! It'll cover the most common case for local dev though